transports/xmpp/src/main/java/org/mule/transport/xmpp/XmppMessageDispatcher.java

```diff
@@ -80,6 +80,12 @@ protected MuleMessage doSend(MuleEvent event) throws Exception
 
     protected void sendMessage(MuleEvent event) throws Exception
     {
+        // Handle session closing by the server
```

Comment:
The receiver uses a Smack listener. For that case, Smack reconnects automatically.